Assets/Scripts/Minigames/MinigameManager.cs

```diff
@@ -32,33 +36,118 @@ public enum MinigameID {
+                minigameTable[kizunaActiveMinigameId].gameObject.SetActive(false);
+                return;
+            case ValidPlayerState.Who.PARTNER:
+                EventManager.InvokeEvent(EventManager.Event.PARTNER_MINIGAME_PAUSE);
+                if (partnerActiveMinigameId != MinigameID.NONE) {
+                    Debug.LogError("There cannot be two minigames active simultaneously");
+                }
+                minigameTable[partnerActiveMinigameId].gameObject.SetActive(false);
+                return;
         }
     }
 
-    public void StartMinigame(MinigameID minigameID, CharacterSwitcher.Who gameOwner) {
-        EventManager.InvokeEvent(EventManager.Event.MINIGAME_START);
-        if (activeMinigameID != MinigameID.NONE) {
-            Debug.LogError("There cannot be two minigames active simultaneously");
+    // Invoked when there is an ongoing minigame for the character that is being switched to
+    public void ResumeMinigame(object inputParam = null) {
+        ValidPlayerState.Who resumingCharacter = StateManager.instance.GetCurrentIdentity();
+
+        switch (resumingCharacter) {
+            case ValidPlayerState.Who.KIZUNA:
+                EventManager.InvokeEvent(EventManager.Event.KIZUNA_MINIGAME_RESUME);
+                if (kizunaActiveMinigameId != MinigameID.NONE) {
+                    Debug.LogError("There cannot be two minigames active simultaneously");
+                }
+                minigameTable[kizunaActiveMinigameId].gameObject.SetActive(true);
+                return;
+            case ValidPlayerState.Who.PARTNER:
+                EventManager.InvokeEvent(EventManager.Event.PARTNER_MINIGAME_RESUME);
+                if (partnerActiveMinigameId != MinigameID.NONE) {
+                    Debug.LogError("There cannot be two minigames active simultaneously");
+                }
+                minigameTable[partnerActiveMinigameId].gameObject.SetActive(true);
+                return;
         }
-        minigameTable[minigameID].SetActive(true);
-        activeMinigameID = minigameID;
-        Minigame minigame = minigameTable[minigameID].GetComponent<Minigame>();
-        minigame.gameOwner = gameOwner;
+
+        Debug.LogError("Invalid character state");
     }
 
     public void ExitMinigame(object inputParam = null) {
-        EventManager.InvokeEvent(EventManager.Event.MINIGAME_END);
-        // NOTE: Can be more defensive and check
-        minigameTable[activeMinigameID].SetActive(false);
-        minigameTable[activeMinigameID].GetComponent<Minigame>().gameOwner = CharacterSwitcher.Who.None;
-        activeMinigameID = MinigameID.NONE;
+        ValidPlayerState.Who exitingCharacter = StateManager.instance.GetCurrentIdentity();
+
+        switch (exitingCharacter) {
+            case ValidPlayerState.Who.KIZUNA:
+                EventManager.InvokeEvent(EventManager.Event.KIZUNA_MINIGAME_END);
+                minigameTable[kizunaActiveMinigameId].gameObject.SetActive(false);
```

Comment:
Maybe should check for exception `kizunaActiveMinigameId == MinigameID.NONE` too same as `PauseMiniGame`.